interview-questions.md

```diff
@@ -9,42 +9,41 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: this is a method for pointing to objects within a class.
 
-  Researched answer:
+  Researched answer: 'This' refers to the objects it refers to. `this`  references the object.. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is the wireframe used to create websites. 
 
-  Researched answer:
+  Researched answer: React is a javascript library for creating user interfaces. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render() is the lifecycle method needed in a react class component. 
 
-  Researched answer:
+  Researched answer: render()
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: javascript extension is the method that allows for html to be written within javascript. 
 
-  Researched answer:
+  Researched answer: Javascript requires parent elements be returned (<></>) to be compilted correctly. In HTML there is no such requirement.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Yarn is a program that installs other programs through the command prompt. The source files...? (not sure)
 
+  Researched answer: Yarn is a package manager for javascript that creates the filestructure of a react app and manages the dependencies. The source files are the ones modified when running yarn.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.